### DISCUSSION TITLE
How do I get the creator of a file using the Github API?

### DISCUSSION THREAD

**Discussion by markcellus** [community/community#24689] [Apps, API and Webhooks]
<p>Hello! I am using the [Repo Commits API](<a href="http://The" rel="nofollow noopener">http://The</a>%20Repo Commits API) to get a list of commits. How do I find out the github user who first committed the file (the creator)?  Thanks in advance!</p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/mkay581">@mkay581</a>,</p>
<p>This API supports the same qualifiers as <a href="http://GitHub.com" rel="nofollow noopener">GitHub.com</a>. To learn more about the format of the query, we recommend checking out these guides:</p>
<ul>
<li><a href="https://developer.github.com/v3/search/#constructing-a-search-query" rel="nofollow noopener">https://developer.github.com/v3/search/#constructing-a-search-query</a></li>
<li><a href="https://help.github.com/en/articles/searching-issues-and-pull-requests" rel="nofollow noopener">https://help.github.com/en/articles/searching-issues-and-pull-requests</a></li>
</ul>
<p>If it helps, we also expose an API for searching commits:</p>
<p><a href="https://developer.github.com/v3/search/#search-commits" rel="nofollow noopener">https://developer.github.com/v3/search/#search-commits</a></p>
<p>That endpoint has an author email and committer email qualifier that might be helpful:</p>
<p><a href="https://help.github.com/en/articles/searching-commits#search-by-author-or-committer" rel="nofollow noopener">https://help.github.com/en/articles/searching-commits#search-by-author-or-committer</a></p>

**Comment by markcellus** [comment]
<p>Thanks for the information and this is slightly helpful but still doesn’t answer my question. How would I find the creator of a file by using the <a href="https://developer.github.com/v3/search/" rel="nofollow noopener">Search API</a> to search commits? Or am I missing something? :thinking:</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>Hey <a class="mention" href="/mkay581">@mkay581</a>,</p>
<p>You can try:</p>
<pre><code>GET /repos/:owner/:repo/commits
</code></pre>
<p>for example:</p>
<p><a href="https://api.github.com/repos/atom/atom/commits">https://api.github.com/repos/atom/atom/commits</a></p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/6/65ab44aed66313b102bbf4d165330c20480dbdcb.png" data-download-href="/uploads/short-url/evpaeNzMIDF40VeTWhSt5EtvmMX.png?dl=1" title="Screen Shot 2019-10-01 at 5.15.09 PM.png"><img src="https://user-images.githubusercontent.com/20666190/181086265-5c330836-196a-47ea-8442-6fc091366fdd.png" alt="Screen Shot 2019-10-01 at 5.15.09 PM.png" data-base62-sha1="evpaeNzMIDF40VeTWhSt5EtvmMX" width="690" height="120" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/6/65ab44aed66313b102bbf4d165330c20480dbdcb_2_690x120.png, https://aws1.discourse-cdn.com/github/optimized/2X/6/65ab44aed66313b102bbf4d165330c20480dbdcb_2_1035x180.png 1.5x, https://aws1.discourse-cdn.com/github/original/2X/6/65ab44aed66313b102bbf4d165330c20480dbdcb.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/6/65ab44aed66313b102bbf4d165330c20480dbdcb_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-10-01 at 5.15.09 PM.png</span><span class="informations">1038×181 13.4 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Whereas, the first committer is Lee Dohm.</p>
<p>Or, on GitHub enterprise:</p>
<p>Use the <a href="https://developer.github.com/v3/repos/commits/#list-commits-on-a-repository">Commits API</a> to list committer information for the first commit.</p>
<p><em>Be sure to adjust the <code>ORG</code> and <code>REPO</code> variables </em></p>
<pre><code>curl -u LOGIN:TOKEN https://HOSTNAME/api/v3/repos/ORG/REPO/commits | jq '[last(.[])] | .[] | .committer.login, .committer.html_url'
</code></pre>
<p>For example, using <code>SuperDevs/RepoForTestOne</code> it would be:</p>
<pre><code>curl -u LOGIN:TOKEN https://HOSTNAME/api/v3/repos/SuperDevs/RepoForTestOne/commits | jq '[last(.[])] | .[] | .committer.login, .committer.html_url'
</code></pre>
<p><strong>Example results:</strong></p>
<p><code>testone</code> &lt;------------- The login of the user who made the first commit on the repository.</p>

**Comment by mudlabs** [comment]
<p>Doesn’t this tell you who was the first user to commit anything to the repo, not who first committed a given file in the repository?</p>
